src/overlays/actors/ovl_En_Fr/z_en_fr.c

```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+                this->scale = 150.0f;
+                this->growingScaleIndex++;
+                if (this->growingScaleIndex >= 4) {
+                    this->growingScaleIndex = 3;
+                }
+                this->isGrowing = false;
+            }
+            break;
+    }
+}
+
+void EnFr_ButterflyPath(EnFr* this, GlobalContext* globalCtx) {
+    s16 rot_y;
+    f32 temp_sin;
+    Vec3f vec1;
+    Vec3f vec2;
+
+    rot_y = this->actor.shape.rot.y;
+
+    this->xyAngleButterfly += 0x1000;
+    Matrix_Translate(this->actor.posRot.pos.x, this->actor.posRot.pos.y, this->actor.posRot.pos.z, MTXMODE_NEW);
+    Matrix_RotateRPY(this->actor.posRot.rot.x, this->actor.posRot.rot.y, this->actor.posRot.rot.z, MTXMODE_APPLY);
+
+    vec1.y = 0.0f;
+    vec1.x = 0.0f;
+    vec1.z = 25.0f;
+    Matrix_MultVec3f(&vec1, &vec2);
+
+    temp_sin = Math_Sins(this->xyAngleButterfly * 2) * 5.0f;
+
+    this->posButterfly.x = (Math_Sins(rot_y) * temp_sin) + vec2.x;
+    this->posButterfly.y = (2.0f * Math_Coss(this->xyAngleButterfly)) + (this->posLogSpot.y + 50.0f);
+    this->posButterfly.z = (Math_Coss(rot_y) * temp_sin) + vec2.z;
+
+    Matrix_Translate(this->posButterfly.x, this->posButterfly.y, this->posButterfly.z, MTXMODE_NEW);
+    Matrix_RotateRPY(this->actor.posRot.rot.x, this->actor.posRot.rot.y, this->actor.posRot.rot.z, MTXMODE_APPLY);
+
+    vec1.x = 0.0f;
+    vec1.y = -15.0f;
+    vec1.z = 20.0f;
+    Matrix_MultVec3f(&vec1, &this->posButterflyLight);
+}
+
+void EnFr_UpdateActive(EnFr* this, GlobalContext* globalCtx) {
+    this->jumpCounter++;
+    Actor_SetScale(&this->actor, this->scale * 0.0001f);
+
+    if (this->isActive) {
+        EnFr_SetScaleActive(this, globalCtx);
+    } else {
+        Actor_SetHeight(&this->actor, 10.0f);
+        this->blinkFunc(this);
+        this->actionFunc(this, globalCtx);
+        EnFr_IsDivingIntoWater(this, globalCtx);
+        EnFr_DivingIntoWater(this, globalCtx);
+        SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+        SkelAnime_FrameUpdateMatrix(&this->skelAnimeButterfly);
+        EnFr_ButterflyPath(this, globalCtx);
+        Actor_MoveForward(&this->actor);
+    }
+}
+
+s32 EnFr_InitJumpingUp(EnFr* this, s32 frogIndex) {
+    EnFr* frog;
+    u8 audio_pitch;
+
+    frog = sEnFrPointers.frogs[frogIndex];
+    if (frog && frog->isJumpingUp == false) {
+        if (frog->growingScaleIndex == 3) {
+            audio_pitch = sPitchLargeFrog[frogIndex];
+        } else {
+            audio_pitch = sPitchSmallFrog[frogIndex];
+        }
+        if (this->songIndex == 6) {
+            frog->isJumpingToFrogSong = true;
+        }
+        frog->isJumpingUp = true;
+        func_800F4BF4(&frog->actor.projectedPos, NA_SE_EV_FROG_JUMP, (s8)audio_pitch);
+        return true;
+    } else {
+        return false;
+    }
+}
+
+void EnFr_Idle(EnFr* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    // If Link pulls out ocarina?
+    if (player->stateFlags2 & 0x2000000) {
```

Comment:
yup. you can remove the comment if you want and itll be clear when we have a macro for the flags